javaharness/java/arcs/android/ArcsService.java

```diff
@@ -22,12 +25,23 @@
   @Inject
   ArcsShellApi arcsShellApi;
 
+  @Inject
+  AndroidArcsEnvironment environment;
+
   @Override
   public void onCreate() {
     super.onCreate();
     Log.d(TAG, "onCreate()");
     DaggerArcsServiceComponent.builder().build().inject(this);
     arcsShellApi.init(this);
+    environment.addReadyListener(recipes -> {
+      recipes.forEach(recipe -> {
+        ArcData arcData = findStartupTrigger(recipe);
+        if (arcData != null) {
+          arcsShellApi.startArc(arcData);
+        }
+      });
+    });
```

Comment:
Please move this part to ArcsShellApi which wraps ArcsEnvironment. ArcsShellApi is the thing wraps the real shell api, which does the actual start up.

Reply:
done